### PR TITLE
Fixed #25212 -- Documented the RawSQL expression.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/25212#ticket

### CODE REVIEW

**Comment by charettes** [comment]
LGTM

**Comment by timgraham** [comment]
Thanks Simon. Since your comment, I added a second commit discouraging the use of `QuerySet.extra()` (since that's dependent on the first one).

File: docs/ref/models/querysets.txt [hunk 1]
```diff
@@ -1120,6 +1120,35 @@ Sometimes, the Django query syntax by itself can't easily express a complex
 ``QuerySet`` modifier — a hook for injecting specific clauses into the SQL
 generated by a ``QuerySet``.
 
+.. admonition:: Use this method as a last resort
+
+    This is an old API that we aim to deprecate at some point in the future.
+    Use it only if you cannot express your query using other queryset methods.
+    If you do need to use it, please `file a ticket
+    <https://code.djangoproject.com/>`_ with your use case so that we can
```
Comment (charettes) [inline docs/ref/models/querysets.txt]: `/newticket`?

**Comment by charettes** [comment]
f1534c8cdb7c6a2d6287082b3e0a6dc9c3f16a39 is also looking good, thanks Tim!

File: docs/ref/models/querysets.txt (hunk 1 above)
Comment (charettes) [inline docs/ref/models/querysets.txt]: Or maybe you're trying to encourage people to search for existing ticket before filing a new one?

Comment (charettes) [inline docs/ref/models/querysets.txt]: What about making sure all tickets about `extra()` use the tag you've started assigning and also suggest having a look at them before filing a new ticket?

Comment (timgraham) [inline docs/ref/models/querysets.txt]: Updated, thanks for the suggestions.

File: docs/ref/models/querysets.txt [hunk 2]
```diff
@@ -1120,6 +1120,37 @@ Sometimes, the Django query syntax by itself can't easily express a complex
...
+        >>> qs.annotate(val=RawSQL("select col from sometable where othercol = %s", (someparam,)))
+
+    The main benefit of using :class:`~django.db.models.expressions.RawSQL` is
+    that you can set ``output_field`` if needed. The main downside is that if
+    you refer to some table alias of the queryset in the raw SQL, then it is
+    possible that Django might change that alias (for example, when the
+    queryset is used as a subquery in yet another query).
+
+    Combining raw SQL with alias relabeling support is possible with the
+    `django-refsql <https://github.com/akaariai/django-refsql>`_ library.
```
Comment (akaariai) [inline docs/ref/models/querysets.txt]: I'm not sure if the refsql library is important enough to be mentioned here.

Otherwise the docs look good.

**Pull request merged**